drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/compiler/OCLBlockVisitor.java

```diff
@@ -222,54 +226,85 @@ private void closeSwitchStatement(Block block) {
         }
     }
 
-    private boolean wasBlockAlreadyClosed(Block b) {
-        Block[] successors = b.getSuccessors();
-        for (Block s : successors) {
-            if (closedLoops.contains(s)) {
-                return true;
-            }
+    private boolean wasBlockAlreadyClosed(Block block) {
```

Comment:
How is the map here? The counter does not mean that a particular block was already closed. Can you explain a bit?

Reply:
It is possible to have multiple nodes that can close a loop (blocks with `LoopExit` nodes). For such cases, we either generate an `else { break; } ` statement or we just close the loop body. To check if a loop has already been closed, I am using the `Map` above to compare against the number of loop exits.